src/Microsoft.AspNetCore.ResponseCaching/ResponseCachingContext.cs

```diff
@@ -107,10 +162,165 @@ private string CreateVaryByCacheKey(CachedVaryBy varyBy)
             return builder.ToString();
         }
 
+        public bool RequestIsCacheable()
+        {
+            // Verify the method
+            // TODO: RFC lists POST as a cacheable method when explicit freshness information is provided, but this is not widely implemented. Will revisit.
+            var request = HttpContext.Request;
+            if (string.Equals("GET", request.Method, StringComparison.OrdinalIgnoreCase))
+            {
+                _responseType = ResponseType.FullReponse;
+            }
+            else if (string.Equals("HEAD", request.Method, StringComparison.OrdinalIgnoreCase))
+            {
+                _responseType = ResponseType.HeadersOnly;
+            }
+            else
+            {
+                return false;
+            }
+
+            // Verify existence of authorization headers
+            // TODO: The server may indicate that the response to these request are cacheable
+            if (!string.IsNullOrEmpty(request.Headers[HeaderNames.Authorization]))
+            {
+                return false;
+            }
+
+            // Verify request cache-control parameters
+            // TODO: no-cache requests can be retrieved upon validation with origin
+            if (!string.IsNullOrEmpty(request.Headers[HeaderNames.CacheControl]))
+            {
+                if (RequestCacheControl.NoCache || RequestCacheControl.NoStore)
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                // Support for legacy HTTP 1.0 cache directive
+                var pragmaHeaderValues = request.Headers[HeaderNames.Pragma];
```

Comment:
More or less. http 1.1 requests might have them for back-compat reasons and when there is no cache control headers, it should be checked instead.